nautobot/core/management/commands/generate_test_data.py

```diff
@@ -80,95 +86,111 @@ def handle(self, *args, **options):
-        self.stdout.write("Creating Aggregates, Prefixes and IP Addresses...")
-        AggregateFactory.create_batch(5, has_tenant_group=True)
-        AggregateFactory.create_batch(5, has_tenant_group=False, has_tenant=True)
-        AggregateFactory.create_batch(10)
-        self.stdout.write("Creating Manufacturers...")
-        ManufacturerFactory.create_batch(10)  # First 10 hard-coded Manufacturers
-        self.stdout.write("Creating Platforms (with manufacturers)...")
-        PlatformFactory.create_batch(20, has_manufacturer=True)
-        self.stdout.write("Creating Platforms (without manufacturers)...")
-        PlatformFactory.create_batch(5, has_manufacturer=False)
-        self.stdout.write("Creating Manufacturers without platforms...")
-        ManufacturerFactory.create_batch(4)  # Remaining 4 hard-coded Manufacturers
-        self.stdout.write("Creating DeviceTypes...")
-        DeviceTypeFactory.create_batch(20)
-        self.stdout.write("Creating DeviceRedundancyGroups...")
-        DeviceRedundancyGroupFactory.create_batch(10)
-        self.stdout.write("Creating CircuitTypes...")
-        CircuitTypeFactory.create_batch(10)
-        self.stdout.write("Creating Providers...")
-        ProviderFactory.create_batch(10)
-        self.stdout.write("Creating ProviderNetworks...")
-        ProviderNetworkFactory.create_batch(10)
-        self.stdout.write("Creating Circuits...")
-        CircuitFactory.create_batch(20)
-        self.stdout.write("Creating Providers without Circuits...")
-        ProviderFactory.create_batch(10)
-        self.stdout.write("Creating CircuitTerminations...")
-        CircuitTerminationFactory.create_batch(2, has_region=True, term_side="A")
-        CircuitTerminationFactory.create_batch(2, has_region=True, term_side="Z")
-        CircuitTerminationFactory.create_batch(2, has_site=False, term_side="A")
-        CircuitTerminationFactory.create_batch(2, has_site=False, term_side="Z")
-        CircuitTerminationFactory.create_batch(2, has_port_speed=True, has_upstream_speed=False)
-        CircuitTerminationFactory.create_batch(
-            size=2,
-            has_site=True,
-            has_location=True,
-            has_port_speed=True,
-            has_upstream_speed=True,
-            has_xconnect_id=True,
-            has_pp_info=True,
-            has_description=True,
-        )
-        # TODO: nautobot.tenancy.tests.test_filters currently calls the following additional factories:
-        # UserFactory.create_batch(10)
-        # RackFactory.create_batch(10)
-        # RackReservationFactory.create_batch(10)
-        # ClusterTypeFactory.create_batch(10)
-        # ClusterGroupFactory.create_batch(10)
-        # ClusterFactory.create_batch(10)
-        # VirtualMachineFactory.create_batch(10)
-        # We need to remove them from there and enable them here instead, but that will require many test updates.
+            self.stdout.write("Creating Roles...")
```

Comment:
Let's think about pulling all this out into a separate function just so we don't have this gigantic `else` block to keep track of.